includes/class-sensei-temporary-user.php

```diff
@@ -120,6 +120,59 @@ public static function filter_learners_query( string $query ) {
 		);
 	}
 
+	/**
+	 * Detect if the e-mail attributes relate to an e-mail from a temporary user.
+	 *
+	 * @access private
+	 * @since  $$next-version$$
+	 *
+	 * @param array  $atts   Email attributes.
+	 * @param string $email_domain The email domain to search for.
+	 * @return boolean Whether to block the email or not.
+	 */
+	public static function should_block_email( $atts, $email_domain ) {
+		$emails = $atts['to'];
+		if ( ! is_array( $emails ) ) {
+			$emails = explode( ',', $emails );
+		}
+		if ( ! empty( $atts['headers'] ) ) {
+			$headers = $atts['headers'];
+			if ( ! is_array( $headers ) ) {
+				// Explode the headers out, so this function can take
+				// both string headers and an array of headers.
+				$temp_headers = explode( "\n", str_replace( "\r\n", "\n", $headers ) );
+			} else {
+				$temp_headers = $headers;
+			}
+			// If it's actually got contents.
+			if ( ! empty( $temp_headers ) ) {
+				foreach ( $temp_headers as $name => $content ) {
+					if ( is_int( $name ) && str_contains( $content, ':' ) ) {
+						list ( $name, $content) = explode( ':', trim( $content ), 2 );
+					}
+
+					// Cleanup crew.
+					$name    = trim( $name );
+					$content = trim( $content );
+
+					if ( in_array( strtolower( $name ), [ 'from', 'cc', 'bcc', 'reply-to' ], true ) ) {
+						$emails = array_merge( (array) $emails, explode( ',', $content ) );
+					}
+				}
+			}
+		}
+		foreach ( $emails as $address ) {
+			if ( preg_match( '/(.*)<(.+)>/', $address, $matches ) && count( $matches ) === 3 ) {
+				$address = $matches[2];
+			}
+			if ( str_ends_with( $address, '@' . $email_domain ) ) {
+				// If this is an e-mail address for a temporary user, don't send it.
+				return true;
+			}
+		}
+		return false;
+	}
+
```

Comment:
This is primarily an attempt on blocking any third party plugins (maybe something related to automation?) from dispatching e-mails (related?) to the temporary user.

We basically look out at the `$to`, `From`, `Cc` , `Bcc` and `Reply-To` headers to look out for any e-mails used by temporary users, and, if there's any reference there, we consider the e-mail "blocked".